source/text/Format.cpp

```diff
@@ -372,6 +373,75 @@ string Format::LowerCase(const string &str)
 
 
 
+// Convert the given non-empty string (noun) to its plural form.
+// Always returns a non-empty string.
+// The actual result might involve some guesswork, and is not guaranteed to be grammatically correct.
+string Format::Plural(const string &noun)
+{
+	if(noun.empty())
+		return noun;
+	static const set<string> exceptionEs={"buffalo", "domino", "echo", "embargo", "hero",
+			"mosquito", "potato", "tomato", "torpedo", "veto"};
+	static const set<char> vowels={'a', 'e', 'i', 'o', 'u', 'y'};
+	const string nounLower = LowerCase(noun);
+	const string lastWord = nounLower.find_last_of(' ') == string::npos
+			? nounLower : nounLower.substr(nounLower.find_last_of(' '), nounLower.size());
+	auto endsWith = [&lastWord](const string &suffix) { return lastWord.size() >= suffix.size()
+			&& 0 == lastWord.compare(lastWord.size() - suffix.size(), suffix.size(), suffix); };
+	if(noun.back() == 's' || endsWith("sh") || endsWith("ch") || endsWith("ch")
+			|| noun.back() == 'x' || noun.back() == 'z')
```

Comment:
Are you just using `.back()` because it's faster if the thing we're looking for is a single character?
I think I'd rather you just call `endsWith()` for everything. The speed is unlikely to matter, if it does, have `endsWith` use a shortcut if it's only one character.
I think it'd also be better to put each of these possible endings into a set and have something step over that set calling `endsWith` with each element.